internal/otelarrow/test/e2e_test.go

```diff
@@ -194,12 +202,16 @@ func testIntegrationTraces(ctx context.Context, t *testing.T, tp testParams, cfg
 
 	expect := make([][]ptrace.Traces, tp.threadCount)
 
+	slp := time.Duration(float64(testCon.expCfg.Arrow.MaxStreamLifetime) * 3 / 2 / float64(tp.requestCount))
 	for num := 0; num < tp.threadCount; num++ {
 		clientDoneWG.Add(1)
 		go func(num int) {
 			defer clientDoneWG.Done()
 			generator := mkgen()
 			for i := 0; i < tp.requestCount; i++ {
+				// Make this process take 1.5x the stream lifetime
+				time.Sleep(slp)
```

Comment:
is it possible to wait on a given condition, instead of waiting on time?

Reply:
Yes. Fixed!
The tests in this file use a callback to signal the end of the test.